### DISCUSSION TITLE
Update minimal part of a range version according some rules

### DISCUSSION THREAD

**Discussion by Michenux** [renovatebot/renovate#30651] [Suggest an Idea]
### Tell us more.

Hi, 

I would like to find a way to update the minimal part of a range version for my terraform modules( example: required_version = ">= 1.5.0" ).
I think there is a bunch of rules that could be used to decide which version to pick but the one i like the most would be to take the oldest version still under support using the datasource https://endoflife.date/.
Other possible rules : a static version, last_3_minors, last_12_months, ...

**Comment by rarkins** [comment]
We've never looked into this substantially but conceptually this concept is:
- Bumping the "lower bounds" of a constraint
- Utilizing "deprecation" data (where we consider non-supported versions to be deprecated)
